### ISSUE TITLE
samples didn't carry out any actions !

### ISSUE THREAD

**Issue by nadamnr** [cuckoosandbox/cuckoo#951]
I have a number of samples which didn't carry out any actions during the analysis process and from the malware page in a number of websites it seems that this is not the normal behaviour !
I have the latest version of cuckoo and these are some examples of the md5s: 
de7747c2ca3fa1f0b75e3d6a910f34e2
e7b0e4d78e958cd5c90c262be459e483
edaae4e8e0068d60f4004f9f36493790
fa9b7555a7fb23a4cc3779500d814580

and there are more but these are just an example 
all the samples belong to 'W23.Sality.AB' family.

**Comment by jbremer** [comment]
I'll give them a spin later today, thanks for reporting.

**Comment by nadamnr** [comment]
Just want to check wheather you had time to look at the samples.

**Comment by jbremer** [comment]
They indeed provide no dynamic analysis. A quick static analysis shows a bunch of obfuscated assembly - I'd have to put some more serious effort to figure out what these samples are doing in the first place. Right now unfortunately I don't have time for that myself, sorry.

**Comment by SparkyNZL** [comment]
I can have a look at these samples if you like ?

**Comment by jbremer** [comment]
If you will, @SparkyNZL, thanks ;-)

**Comment by SparkyNZL** [comment]
can you point me in the direction on where i can download these samples ? Ill run these up on a bare-metal. nadamnr, what is your configuration you are using with cuckoo, eg virtualization environment.

**Comment by nadamnr** [comment]
Sorry, just saw your reply .. 
I am using Virtualbox last version on Ubuntu.
I downloaded the samples from virustotal website maybe you can find them as well on malwr.com 
however, if you couldn't find them I'll be happy to upload them to you.

**Comment by nadamnr** [comment]
It seems that the sample try to open the following registry key with 'NtOpenKey': 
HKEY_LOCAL_MACHINE\\Software\\Microsoft\\Windows NT\\CurrentVersion\\Image File Execution Options\\kernel32.dll
and when this returned : 3221225524
the sample executed .. 
does this key has anything to do with detecting a virtualise environment or something ?

**Comment by joshuachad** [comment]
I'm not sure it does but the code ="object name not found". I assume this
will not be modified by virtual
Machine software but perhaps by a dugger.

**Comment by nadamnr** [comment]
I see,  thank you for your reply .. Then can the malware check whether its running on a sandbox or not without this being monitoted on such sandbox. . is there known ways to do that ?

**Comment by SparkyNZL** [comment]
Loads of malware can determine if its operating in a virtual environment, The detection of the VMX string use to be a method a while back, many authors would obfuscate this string in the application but once loaded into memory you could pick it up. Other ways were detecting directory structures VMware tools installation and the list goes on :( but its always fun learn the new ways these people are using to hide their detection mechanisms.

**Comment by SparkyNZL** [comment]
oh , do you want me to have a look at these sample for you still ? i don't mind IDA and Olly preferred tools along with a few others.

**Comment by nadamnr** [comment]
I sure do , can I upload them here ? Of course as a password protected compressed file !

But commenting on your 'comment', can malware , in general,  detect that they are running on a sandbox without calling any API = Without any behavioural actions !

**Comment by SparkyNZL** [comment]
sure zip them up and send them to sparkie at borty.co.nz

**Comment by SparkyNZL** [comment]
Yes, I have seen ones which have all the detection code in the malware itself rather than calling an API from say a DLL to deliver it information about the environment. But the malware would still have to call APIs before it got to that point. (am i understanding you correctly ?) e.g. To even start to execute it needs to call API's to setup its environment. But then this gets into an argument about what an API actually is..

**Comment by SparkyNZL** [comment]
Hi Madamnr , did you email the samples through ?

**Comment by nadamnr** [comment]
I'll do today hopefully.. Sorry I was away from home for sometime.

**Comment by Lindi16** [comment]
new progress on the samples?
-need help?
--> i'm allways intrested to learn how, malware detect the sandbox -> and then harden mine against this tec 🎉

**Comment by SparkyNZL** [comment]
Hi  nadamnr ,

I have had a look at sample : ba5af06e75b0936588809832901da03970cfe5204a0f6afbfbdcb749e6732866

This sample is very basic, and as far as i can see its not deletecting the Virtural Envirement.

So what do i think this.

1: Behaviour is similar on VMware and Virtualbox
2: No VMX string detection etc.
3: Imports table has only 2 API's both from Kernel32
      - seterrormode
      - exitprocess

What i can tell you its it looks like the binary is looking for arguments, as a hunch i would think that this was probably one of a number of files dropped..

The second sample im still looking at, it has 51 imports fro 3 different DLL;s

Cheers
G

**Comment by nadamnr** [comment]
Many thanks :), 
waiting eagerly for the second one.
just a question ... how can I identify whether there is a 'VMX string detection' ?
